### ISSUE TITLE
Can't mount volume with docker-compose

### ISSUE THREAD

**Issue by vladyslavstartsev** [phpmyadmin/docker#181]
I use `laradock`.
And I can't mount  - `./phpmyadmin/config/config.user.inc.php:/etc/phpmyadmin/config.user.inc.php`
Looked [here](https://docs.phpmyadmin.net/en/latest/setup.html?highlight=docker-compose#customizing-configuration-file-using-docker-compose)
my `docker-compose.yml` phpmyadmin section looks like this
```yml
phpmyadmin:
  build: ./phpmyadmin
  volumes:
    - ./phpmyadmin/config/config.user.inc.php:/etc/phpmyadmin/config.user.inc.php
  environment:
    - PMA_ARBITRARY=1
    - MYSQL_USER=${PMA_USER}
    - MYSQL_PASSWORD=${PMA_PASSWORD}
    - MYSQL_ROOT_PASSWORD=${PMA_ROOT_PASSWORD}
  ports:
    - "${PMA_PORT}:80"
  depends_on:
    - "${PMA_DB_ENGINE}"
  networks:
    - frontend
    - backend
```
I do `docker-compose build --no-cache phpmyadmin` and then `docker-compose up -d phpmyadmin`. What it does is creates `phpmyadmin/config/config.user.inc.php` folder (!!!)

if I delete `phpmyadmin/config` folder, change `docker-compose.yml` phpmyadmin section to 
```yml
volumes:
  - ./phpmyadmin/config:/etc/phpmyadmin

```
do `docker-compose build --no-cache phpmyadmin` and then `docker-compose up -d phpmyadmin`, it will create `phpmyadmin/config` with files from container. BUT
1. My `phpmyadmin` login page changes 
from 
![image](https://user-images.githubusercontent.com/17382248/42897436-b487a234-8ac8-11e8-86d7-550c4fc5ea9d.png)

to 
![image](https://user-images.githubusercontent.com/17382248/42897479-d02fb95e-8ac8-11e8-923d-2d0d0db109ec.png)

Server input row is dropped

2. If I will try to change something there, for example I've tried to put `var_dump('random text');die;` in `phpmyadmin/config/config.user.inc.php` in will not be executed. At least if I visit `localhost:8080` (`laradock` default route for `phpmyadmin`) I will not see my `random text` in page source.

**Comment by danielecr** [comment]
if the file or folder does not exists docker daemon will create that assuming it is a folder, so

```
  volumes:
    - ./phpmyadmin/config/config.user.inc.php:/etc/phpmyadmin/config.user.inc.php
```
will work if the file ./phpmyadmin/config/config.user.inc.php exists before docker-compose is runned by 'up', and dockerd is directed to run the container that way.

Using

```
volumes:
  - ./phpmyadmin/config:/etc/phpmyadmin
```

will replace the container folder /etc/phpmyadmin completely, so no config.inc.php is readed, and config.user.inc.php is included from config.inc.php, so any changes has no effect.
You can optionally paste the config.inc.php from source in
https://raw.githubusercontent.com/phpmyadmin/docker/master/etc/phpmyadmin/config.inc.php
or just create the file config.user.inc.php before docker-compose up

and server selection disappears because it is readed from environment variables from config.inc.php, and that file, from the container point of view, does not exists anymore.

**Comment by williamdes** [comment]
@danielecr Is this still an issue on latest version ?
Looks like #243

**Comment by danielecr** [comment]
> @danielecr Is this still an issue on latest version ?
> Looks like #243

Sorry, I can not check now, I setted a remind for the next week

**Comment by williamdes** [comment]
@danielecr :ping_pong: did you have time to test ?

**Comment by danielecr** [comment]
> @danielecr  did you have time to test ?

I can test with PMA_ARBITRARY=1 in docker-compose, and it works fine.
What I described there is the normal/wanted behavior of docker daemon: if mount source does not exists, dockerd assumes it is a folder, then it create it. And of course mounting a folder over existing container target folder will hide the original (packed) content of the folder.

Nothing to add from my side, @vladyslavstartsev can say more maybe, for convenience I tested the example he proposed, writing 

```
<?php
var_dump('random text');die;
```
into a `config.user.inc.php` mounted as `/etc/phpmyadmin/config/config.user.inc.php`
then
~~~
string(11) "random text"
~~~
is the only thing printed by the server, as expected.

I do not think there is nothing more to test or discuss further
